File: class22/python/AP2203-Exemplo02.py
# FuncaoFatorial — Aula 22
# Função que calcula o fatorial de um número

def fatorial(n):
    if n < 0:
        return None
    resultado = 1
    for i in range(2, n + 1):
        resultado *= i
    return resultado

num = int(input("Digite um número: "))
fat = fatorial(num)
if fat is None:
    print("Não existe fatorial de número negativo!")
else:
    print(f"{num}! = {fat}")

File: class22/cpp/AP2203-Exemplo02.cpp
#include <iostream>
long long fatorial(int n) {
    if (n < 0) return -1;
    long long fat = 1;
    for (int i = 2; i <= n; i++) fat *= i;
    return fat;
}
int main() {
    int n;
    std::cout << "Digite um número: "; std::cin >> n;
    long long fat = fatorial(n);
    if (fat == -1) std::cout << "Não existe fatorial de número negativo!" << std::endl;
    else std::cout << n << "! = " << fat << std::endl;
    return 0;
}